### PR TITLE
[Costing] 'newtype' 'intercept' and 'slope'

### PR DESCRIPTION
**Pull request by effectfully**
To make sure we don't confuse the two.

### CODE REVIEW

**Comment by effectfully** [comment]
Needs a changelog entry, otherwise ready for review.

**Review by michaelpj — APPROVED** [review]

File: plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostingFun/Core.hs [hunk 1]
```diff
@@ -177,51 +192,51 @@ runOneArgumentModel (ModelOneArgumentLinearCost (ModelLinearSize intercept slope
 
 -- | s * (x + y) + I
 data ModelAddedSizes = ModelAddedSizes
-    { modelAddedSizesIntercept :: CostingInteger
-    , modelAddedSizesSlope     :: CostingInteger
+    { modelAddedSizesIntercept :: Intercept
```
Comment (kwxm) [inline plutus-core/plutus-core/src/PlutusCore/Evaluation/Machine/CostingFun/Core.hs:195]: Maybe we should make all of these model types have a sub-record that describes a linear function via named fields or something, but let's think about that some other time.  This solve the problem for the time being.

**Review by kwxm — APPROVED** [review]
Yep, looks fine.

**Comment by kwxm** [comment]
So it's _not_ just me that's getting these "Unexpected end-of-file" errors.

**Comment by effectfully** [comment]
Oh, what's that? Repushing.

**Comment by effectfully** [comment]
@michaelpj any idea what's going on with CI?

**Comment by michaelpj** [comment]
Weird issue that other people have been seeing too, I'm just restarting the jobs for now.

**Pull request merged**